app/models/patient.rb

```diff
@@ -142,6 +166,24 @@ def self.contacted_since(datetime)
       first_contacts: first_contact.length, pledges_sent: 20 }
   end
 
+  def self.to_csv
+    CSV.generate(encoding: 'utf-8') do |csv|
+      csv << CSV_EXPORT_FIELDS.keys # Header line
+      all.each do |patient|
+        csv << CSV_EXPORT_FIELDS.values.map do |field|
+          value = patient.public_send(field)
```

Comment:
Oh durf, the public send thing choking on has_alt_contact totally makes sense! My mistake, let's definitely go with `public_send` in that case.

re: `special_circumstances`, an unoptimized facet of the way we're saving them now is that we save a couple blank strings. It's an inconvenience but it's not causing any particular harm and we don't generate data that quickly, so it's on the backlog down a ways. But great eye noting that for sure, and great call working around it.

I really love the `is_a?` option you present here. It's really solid and crisp, and super clear what it's checking for! Would love to go with that, if it's cool with you.